Presentation/UIViewController+Presentation.swift

```diff
@@ -26,6 +26,7 @@ public extension UIViewController {
             }
         }
 
+        let previous = (self as? UINavigationController)?.isNavigationBarHidden
```

Comment:
This working only if `present` is called from a `UINavigationController` but not from a `UIViewController` contained in a `UINavigationController`.
Let's say we have 
```swift
let nc = UIViewController()
let vc1 = UIViewController()
let vc2 = UIViewController()
nc.present(vc1)
vc1.present(vc2) // here vc2 will be pushed into nc, even though present is called on vc1
```

Reply:
This is managed in UINavigationController+Presenting